src/NHibernate/Linq/Visitors/HqlGeneratorExpressionVisitor.cs

```diff
@@ -226,18 +226,14 @@ private HqlTreeNode VisitNhNominated(NhNominatedExpression nhNominatedExpression
 
 		private HqlTreeNode VisitInvocationExpression(InvocationExpression expression)
 		{
-			//This is an ugly workaround for dynamic expressions.
-			//Unfortunately we can not tap into the expression tree earlier to intercept the dynamic expression
-			if (expression.Arguments.Count == 2 &&
-			    expression.Arguments[0] is ConstantExpression constant &&
-			    constant.Value is CallSite site &&
-			    site.Binder is GetMemberBinder binder)
+#if NETCOREAPP2_0
```

Comment:
Why NETCOREAPP2_0 only?

Reply:
Because in .NET Framework a dynamic expression [will never be reduced](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/DynamicExpression.cs#L147) with an `ExpressionVisitor`, where in .NET Core [it will be reduced](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/DynamicExpression.cs#L149) when the visitor is not a `DynamicExpressionVisitor`.